src/main/kotlin/uk/gov/dluhc/printapi/service/SftpService.kt

```diff
@@ -51,4 +53,19 @@ class SftpService(
         sftpOutboundTemplate.rename("$fileDirectoryPath/$originalFileName", "$fileDirectoryPath/$newFileName")
         return newFileName
     }
+
+    /**
+     * Fetches the file from the path on the remote server and
+     * unmarshalls the content to an object of type `responseObjectType`
+     * @param filePathToProcess the path to the file on the remote server
+     * @param responseObjectType the type of the object to marshall the json content of the file to
+     */
+    fun <T> fetchAndUnmarshallFile(filePathToProcess: String, responseObjectType: Class<T>): T {
```

Comment:
Kinda related to my previous comment, I dont know if its right that the SFTP service is responsible for deserializing into an object .... 🤔 
I know you've used generics, which means it could deserialize to any type, but might it be better if this method simply returned a string, then the thing that called it can worry about turning it into a thing that is relevant to that class/use case
(which would mean we _do_ need the method `fetchAndUnmarshallPrintResponses` in `PrintResponseFileService`)

(Also .... very minor naming question .... but is "unmarshall" the right terminology? To me (and I could be alone in this) "marshalling" and "unmarshalling" makes me think of XML and jaxb ??)